strawberry/experimental/pydantic/fields.py

```diff
@@ -106,24 +109,26 @@ def replace_types_recursively(type_: Any, is_input: bool) -> Any:
     basic_type = get_basic_type(type_)
     replaced_type = replace_pydantic_types(basic_type, is_input)
 
-    origin = getattr(type_, "__origin__", None)
-    if origin is Literal:
-        # Literal does not have types in its __args__ so we return early
+    origin = get_origin(type_)
+    if not origin or not hasattr(type_, "__args__"):
         return replaced_type
-    if hasattr(replaced_type, "__args__"):
-        replaced_type = replaced_type.copy_with(
-            tuple(
-                replace_types_recursively(t, is_input=is_input)
-                for t in replaced_type.__args__
-            )
+
+    converted = tuple(
+        replace_types_recursively(t, is_input=is_input) for t in get_args(replaced_type)
+    )
+
+    if isinstance(replaced_type, TypingGenericAlias):
+        return TypingGenericAlias(origin, converted)
+
+    replaced_type = replaced_type.copy_with(converted)
+
+    if isinstance(replaced_type, TypeDefinition):
+        # TODO: Not sure if this is necessary. No coverage in tests
+        # TODO: Unnecessary with StrawberryObject
```

Comment:
@silas @thejaminator 
I just wonder this code.
Why does the TypeDefinition check logic exist? In what cases does this code work?
Referring to the comments, is the logic wriiten for any error(defense)?